### PR TITLE
Feat: Create new requests for event service

### LINKED ISSUE #62: Fix: handle optional fields correctly in proto
Protobuf sets a default value for null such as:
* "" for string
* 0 for int
* 1970-01-01 for datetime
Handle it

### PR DESCRIPTION
**Pull request by ZixMai**
1.  Added delete events by author id request
2.  Added search by timestamp in get events by author id and get all events (With SQL Query)
3.  Renamed old delete method to delete event by id
4.  Fixed pagination and delete methods in mock
5.  Added WrongIntervalError
6.  Added Interval type class
7.  Changed type of repeating_delay field from datetime to Interval
8.  Changed optional fields that have wrong default value to WhichOneOf
9.  Added converters for Interval type
10.  Added check for wrong interval in create and update event methods

### CODE REVIEW

File: event_service/repository/event_repository_impl.py [hunk 1]
```diff
@@ -25,20 +25,22 @@ class EventRepositoryImpl(EventRepositoryInterface):
...
-    async get_events_by_author_id(author_id, page_number, items_per_page)
-        Returns page with events that has matches with given author id.
+    async get_events_by_author_id(author_id, page_number, items_per_page, start, end)
+        Returns page with events those have matches with given author id.
     async get_event_by_event_id(event_id)
         Returns event that has matches with given event id.
     async get_events_by_events_ids(events_ids, page_number, items_per_page)
-        Returns page of events that has matches with given list of event ids.
-    async get_all_events(page_number, items_per_page)
+        Returns page of events those have matches with given list of event ids.
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: that was right in docstring

File: event_service/repository/event_repository_impl.py [hunk 2]
```diff
@@ -25,20 +25,22 @@ class EventRepositoryImpl(EventRepositoryInterface):
 
     Methods
     -------
-    async get_events_by_author_id(author_id, page_number, items_per_page)
-        Returns page with events that has matches with given author id.
+    async get_events_by_author_id(author_id, page_number, items_per_page, start, end)
+        Returns page with events those have matches with given author id.
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: that was right in docstring

File: event_service/repository/event_repository_impl.py [hunk 3]
```diff
@@ -61,11 +68,15 @@ async def get_events_by_author_id(
...
+        start : Optional[datetime]
+            Start of time interval for search.
+        end : Optional[datetime]
+            End of time interval for search.
 
         Returns
         -------
         List[Event]
-            List of events that matches by author id.
+            List of events those match by author id.
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: that was right in docstring

File: event_service/repository/event_repository_impl.py [hunk 4]
```diff
@@ -78,7 +89,18 @@ async def get_events_by_author_id(
...
+                "author_id": author_id,
+                "deleted_at": None,
+            }
+            | (
+                {}
+                if not (
+                    timestamp := ({} if start is None else {"_gte": start})
+                    | ({} if end is None else {"_lte": end})
+                )
+                else {"start": timestamp}
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: what about end?

File: event_service/repository/event_repository_impl.py [hunk 5]
```diff
@@ -135,7 +157,7 @@ async def get_events_by_events_ids(
         Returns
         -------
         List[Event]
-            List of events that matches by event id.
+            List of events those match by event id.
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: that was right in docstring

File: event_service/repository/event_repository_impl.py [hunk 6]
```diff
@@ -190,6 +220,12 @@ async def get_all_events(
         db_events: Optional[
             List[PrismaEvent]
         ] = await self._db_client.db.event.find_many(
+            where={}
+            if not (
+                timestamp := ({} if start is None else {"_gte": start})
+                | ({} if end is None else {"_lte": end})
+            )
+            else {"start": timestamp},
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: where is end?

File: event_service/repository/event_repository_interface.py [hunk 7]
```diff
@@ -12,29 +13,36 @@ class EventRepositoryInterface(ABC):
 
     Methods
     -------
-    async get_events_by_author_id(author_id, page_number, items_per_page)
-        Returns page with events that has matches with given author id.
+    async get_events_by_author_id(author_id, page_number, items_per_page, start, end)
+        Returns page with events those have matches with given author id.
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_interface.py]: that was right in docstring

File: event_service/repository/event_repository_interface.py [hunk 8]
```diff
@@ -12,29 +13,36 @@ class EventRepositoryInterface(ABC):
...
     async get_event_by_event_id(event_id)
         Returns event that has matches with given event id.
     async get_events_by_events_ids(events_ids, page_number, items_per_page)
-        Returns page of events that has matches with given list of event ids.
-    async get_all_events(page_number, items_per_page)
+        Returns page of events those have matches with given list of event ids.
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_interface.py]: that was right in docstring

File: event_service/repository/event_repository_interface.py [hunk 9]
```diff
@@ -44,11 +52,15 @@ async def get_events_by_author_id(
...
+        start : Optional[datetime]
+            Start of time interval for search.
+        end : Optional[datetime]
+            End of time interval for search.
 
         Returns
         -------
         List[Event]
-            List of events that matches by author id.
+            List of events those match by author id.
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_interface.py]: that was right in docstring

File: event_service/repository/mock_event_repository.py [hunk 10]
```diff
@@ -22,20 +23,22 @@ class MockEventRepositoryImpl(EventRepositoryInterface):
 
     Methods
     -------
-    async get_events_by_author_id(author_id, page_number, items_per_page)
-        Returns page with events that has matches with given author id.
+    async get_events_by_author_id(author_id, page_number, items_per_page, start, end)
+        Returns page with events those have matches with given author id.
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py]: that was right in docstring

File: event_service/repository/mock_event_repository.py [hunk 11]
```diff
@@ -22,20 +23,22 @@ class MockEventRepositoryImpl(EventRepositoryInterface):
...
     async get_event_by_event_id(event_id)
         Returns event that has matches with given event id.
     async get_events_by_events_ids(events_ids, page_number, items_per_page)
-        Returns page of events that has matches with given list of event ids.
-    async get_all_events(page_number, items_per_page)
+        Returns page of events those have matches with given list of event ids.
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py]: that was right in docstring

File: event_service/repository/mock_event_repository.py [hunk 12]
```diff
@@ -58,11 +66,15 @@ async def get_events_by_author_id(
...
+        start : Optional[datetime]
+            Start of time interval for search.
+        end : Optional[datetime]
+            End of time interval for search.
 
         Returns
         -------
         List[Event]
-            List of events that matches by author id.
+            List of events those match by author id.
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py]: that was right in docstring

File: event_service/repository/mock_event_repository.py [hunk 13]
```diff
@@ -130,7 +148,7 @@ async def get_events_by_events_ids(
         Returns
         -------
         List[Event]
-            List of events that matches by event id.
+            List of events those match by event id.
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py]: that was right in docstring

File: event_service/src/event_service_impl.py [hunk 14]
```diff
@@ -37,8 +39,10 @@ class EventServiceImpl(GrpcServicer):
         Function that need to be bind to the server that creates the event.
     async update_event(request, context)
         Function that need to be bind to the server that updates the event.
-    async delete_event(request, context)
-        Function that need to be bind to the server that deletes the event.
+    async delete_event_by_id(request, context)
+        Function that need to be bind to the server that deletes the event that matches id.
+    async delete_events_by_author_id(request, context)
+        Function that need to be bind to the server that deletes events those matches id.
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: those should be replaced with that

File: event_service/src/event_service_impl.py [hunk 15]
```diff
@@ -75,6 +79,10 @@ async def get_events_by_author_id(
             author_id=request.author_id,
             page_number=request.page_number,
             items_per_page=request.items_per_page,
+            start=datetime.fromtimestamp(
+                request.start.seconds + request.start.nanos / 1e9
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py:83]: check for none

File: event_service/src/event_service_impl.py [hunk 16]
```diff
@@ -75,6 +79,10 @@ async def get_events_by_author_id(
...
+            ),
+            end=datetime.fromtimestamp(request.end.seconds + request.end.nanos / 1e9),
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: check for none

File: event_service/src/event_service_impl.py [hunk 17]
```diff
@@ -166,7 +174,12 @@ async def get_all_events(
         if request.requesting_user.type != GrpcUserType.ADMIN:
             raise PermissionDeniedError("Permission denied")
         events = await self._event_repository.get_all_events(
-            page_number=request.page_number, items_per_page=request.items_per_page
+            page_number=request.page_number,
+            items_per_page=request.items_per_page,
+            start=datetime.fromtimestamp(
+                request.start.seconds + request.start.nanos / 1e9
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py:189]: check for none

File: event_service/src/event_service_impl.py [hunk 18]
```diff
@@ -166,7 +174,12 @@ async def get_all_events(
...
+            ),
+            end=datetime.fromtimestamp(request.end.seconds + request.end.nanos / 1e9),
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: check for none

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

File: event_service/repository/event_repository_impl.py (hunk 4 above)
Comment (ZixMai) [reply event_service/repository/event_repository_impl.py]: this is normal

File: event_service/repository/event_repository_impl.py (hunk 6 above)
Comment (ZixMai) [reply event_service/repository/event_repository_impl.py]: 2 strings upper

**Review by MaxPlays35 — APPROVED** [review]
LGTM

File: event_service/repository/mock_event_repository.py [hunk 19]
```diff
@@ -245,6 +360,32 @@ async def delete_event(self, event_id: str) -> None:
...
+            Author's id.
+
+        Raises
+        ------
+        ValueNotFoundError
+            Can't delete event with provided data.
+
+        """
+        try:
+            index = next(
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py]: This function should delete all author's events

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Pull request merged**